### ISSUE TITLE
Error after nhl-setup when not simple setup

### ISSUE THREAD

**Issue by Overboard-code** [riffnshred/nhl-led-scoreboard#374]
after trying simple config I wanted to add weather and other screens results were:
```

pi@hockey:~/nhl-led-scoreboard $ sudo python3 src/main.py --led-rows=64 --led-cols=128 --led-gpio-mapping=adafruit-hat-pwm --led-slowdown-gpio=2
INFO: Validating config.json.....
WARN: config.json fails validation: error: [data.preferences.goal_animations must contain ['pref_team_only'] properties]
WARN: Rerun the nhl_setup app to create a valid config.json

```
In the config file it looks like:
```
     "goal_animations": {}
```
it looks like goal animations didn't configure correctly:  I think it should be 
```
       "goal_animations": 
        {
                "pref_team_only": false
        }

```
**Logs**
```
pi@hockey:~/nhl-led-scoreboard $ ./nhl_setup
   _  ____ ____     __   _______
  / |/ / // / /    / /  / __/ _ \
 /    / _  / /__  / /__/ _// // /
/_/|_/_//_/____/ /____/___/____/

   _____________  ___  _______  ____  ___   ___  ___
  / __/ ___/ __ \/ _ \/ __/ _ )/ __ \/ _ | / _ \/ _ \
 _\ \/ /__/ /_/ / , _/ _// _  / /_/ / __ |/ , _/ // /
/___/\___/\____/_/|_/___/____/\____/_/ |_/_/|_/____/

   ________________  _____
  / __/ __/_  __/ / / / _ \
 _\ \/ _/  / / / /_/ / ___/
/___/___/ /_/  \____/_/

nhl_setup V1.6.2
nhl led scoreboard V1.6.9
Now validating config......
Your config.json passes validation and can be used with nhl led scoreboard
🥅🏒🚨 Do you see a net,stick and horn? Yes
🥅 Do you want a simple default setup with one team selection (Y)? No
🥅 What kind of setup do you want? full
🥅 Do you want full debug on (only use this if asked to)? (outputs ALL log messages to console) No
🥅 What log level do you want?  INFO
🥅 Do you want live_mode on? (Shows live game data of favorite team) Yes
🥅 Select the time format 12h
🥅 When do you want end of day to be? 22:00
🥅 Your location to override latitude and longitude lookup via IP (City, State or City, Province or lat,lon or home address)
🥅 Live game refresh rate? (rate at which a live game will call the NHL API to catch the new data, 10 seconds is the lowest) 10
🥅 Select your preferred team: (Use arrow keys)
 » Hurricanes
   Avalanche
   Blackhawks
   Blues
   Blue Jackets
   Bruins
   Canadiens
   Canucks
   Capitals
   Coyotes
   Devils
   Ducks
   Flames
   Flyers
   Golden Knights
   Islanders
   Jets
   Kings
   Kraken
   Maple Leafs
   Lightning
   Oilers
   Panthers
   Penguins
   Predators
   Rangers
   Red Wings
🥅 Select your preferred team: Hurricanes
🥅 Add another team? No
🥅 SOG Display Frequency? (How often do you want to see shots on goal) 4
🥅 Do you want goal animations for only preferred team?
🥅 Select a board to display for off_day clock
🥅 Add another board? Yes
🥅 Select a board to display for off_day weather
🥅 Add another board? Yes
🥅 Select a board to display for off_day wxalert
🥅 Add another board? Yes
🥅 Select a board to display for off_day wxforecast
🥅 Add another board? Yes
🥅 Select a board to display for off_day scoreticker
🥅 Add another board? Yes
🥅 Select a board to display for off_day seriesticker
🥅 Add another board? Yes
🥅 Select a board to display for off_day standings
🥅 Add another board? Yes
🥅 Select a board to display for off_day team_summary
🥅 Add another board? Yes
🥅 Select a board to display for off_day stanley_cup_champions
🥅 Add another board? No
🥅 Select a board to display for scheduled clock
🥅 Add another board? Yes
🥅 Select a board to display for scheduled weather
🥅 Add another board? Yes
🥅 Select a board to display for scheduled wxalert
🥅 Add another board? Yes
🥅 Select a board to display for scheduled wxforecast
🥅 Add another board? Yes
🥅 Select a board to display for scheduled seriesticker
🥅 Add another board? Yes
🥅 Select a board to display for scheduled standings
🥅 Add another board? Yes
🥅 Select a board to display for scheduled team_summary
🥅 Add another board? Yes
🥅 Select a board to display for scheduled scoreticker
🥅 Add another board? No
🥅 Select a board to display for intermission clock
🥅 Add another board? Yes
🥅 Select a board to display for intermission scoreticker
🥅 Add another board? Yes
🥅 Select a board to display for intermission seriesticker
🥅 Add another board? Yes
🥅 Select a board to display for intermission standings
🥅 Add another board? No
🥅 Select a board to display for post_game clock
🥅 Add another board? Yes
🥅 Select a board to display for post_game seriesticker
🥅 Add another board? Yes
🥅 Select a board to display for post_game standings
🥅 Add another board? Yes
🥅 Select a board to display for post_game team_summary
🥅 Add another board? Yes
🥅 Select a board to display for post_game stanley_cup_champions
🥅 Add another board? No
🥅 Duration clock is shown 15
🥅 Hide network indicator when clock displayed (for when there are network issues, red bar on bottom of display) No
🥅 Use your first preferred team's colors for clock and date? Yes
🥅 Flash seconds? Yes
🥅 Use weather data feed (this is required to get data for the weather and weather alert boards)? Yes
🥅 Units to display? (metric or imperial) imperial
🥅 How often to update weather in minutes?(minimum 5) 10
🥅 Weather observation display summary
🥅 How long to show weather board (minimum 30 seconds)? 30
🥅 Which weather data feed for current observations? EC=Environment Canada
OWM=Open Weather Map (requires a key: works for all locations) OWM
🥅 OpenWeatherMap API key if using OWM as data feed: (get key from https://openweathermap.org/appid) 584a45a2e53805b06ec4bb1ac18680ab
🥅 Display temperature and humidity on clock? Yes
🥅 Get weather forecast from your weather provider? Yes
🥅 Number of days forecast to show?(minimum 1, max 3) 3
🥅 How often to update weather forecast in hours?(minimum 1) 1
🥅 Show weather alerts? Yes
🥅 Which weather feed for alerts? EC=Environment Canada
NWS=National Weather Service (US only) NWS
🥅 On alert board, display title of alert (warning, watch, advisory)? Yes
🥅 For NWS alert feed use expire time rather than effective time? Yes
🥅 On alert board, scroll alert? Yes
🥅 How long (in seconds) to show the alert board 5
🥅 How often to update alert feed in minutes?(minimum 5) 5
🥅 Display alert notification (Fred) on clock? Yes
🥅 Score Ticker: Show preferred teams only? (Show only your preferred team or all games of the day) No
🥅 Score Ticker: Board rotation rate? (How often do you want to rotate the games shown) 3
🥅 Series Ticker: Show preferred teams only? (Show only your preferred team or all the series of the playoff) No
🥅 Series Ticker: Board rotation rate? (How often do you want to rotate the series shown) 3
🥅 Show preferred standings only? (Show all standings or your preferred division and conference) Yes
🥅 Select the type of standings to display division
🥅 Select the division to display metropolitan
🥅 Select the conference to display eastern
🥅 Use pushbutton No
🥅 Use dimmer Yes
🥅 Select source of dimmer, software (uses your IP to find sunrise/sunset) or hardware (a sensor attached) software
🥅 Frequency in minutes to check if dimming should happen 5
🥅 When to allow dimming, always or only on off days always
🥅 How bright the display should be at night (between 5 and 100) 5
🥅 How bright the display should be during the day  (between 5 and 100) 40
🥅 Override automatic sunrise/sunset with set times? No
🥅 Add offset in minutes to automatic sunrise/sunset? (+ value adds to time, - subtracts from time Yes
🥅 Offset in minutes to sunrise/sunset 30
🥅 Use screensaver Yes
🥅 Use animations stored in assets/animations/screensaver? Yes
🥅 When do you want to start the screensaver? (24h) 22:00
🥅 When do you want the dimmer to stop the screensaver? (24h) 07:00
🥅 Update data feeds while screensaver is on? (applies to all weather feeds and dimmer, nhl will always be blocked with screensaver on) No
🥅 Use motion sensor to stop/start screensaver? No
🥅🏒🚨 Save config/config.json file? Yes
```

**Your config file**
```
 pi@hockey:~/nhl-led-scoreboard $ cat config/config.json
{
    "debug": false,
    "loglevel": "INFO",
    "live_mode": true,
    "preferences": {
        "time_format": "12h",
        "end_of_day": "22:00",
        "location": "",
        "live_game_refresh_rate": 10,
        "teams": [
            "Hurricanes"
        ],
        "sog_display_frequency": 4,
        "goal_animations": {}
    },
    "states": {
        "off_day": [
            "clock",
            "weather",
            "wxalert",
            "wxforecast",
            "scoreticker",
            "seriesticker",
            "standings",
            "team_summary",
            "stanley_cup_champions"
        ],
        "scheduled": [
            "clock",
            "weather",
            "wxalert",
            "wxforecast",
            "seriesticker",
            "standings",
            "team_summary",
            "scoreticker"
        ],
        "intermission": [
            "clock",
            "scoreticker",
            "seriesticker",
            "standings"
        ],
        "post_game": [
            "clock",
            "seriesticker",
            "standings",
            "team_summary",
            "stanley_cup_champions"
        ]
    },
    "boards": {
        "scoreticker": {
            "preferred_teams_only": false,
            "rotation_rate": 3
        },
        "seriesticker": {
            "preferred_teams_only": false,
            "rotation_rate": 3
        },
        "standings": {
            "preferred_standings_only": true,
            "standing_type": "division",
            "divisions": "metropolitan",
            "conference": "eastern"
        },
        "clock": {
            "duration": 15,
            "hide_indicator": false,
            "preferred_team_colors": true,
            "clock_rgb": "",
            "date_rgb": "",
            "flash_seconds": true
        },
        "weather": {
            "enabled": true,
            "view": "summary",
            "units": "imperial",
            "duration": 30,
            "data_feed": "OWM",
            "owm_apikey": "584a45a2e53805b06ec4bb1ac18680ab",
            "update_freq": 10,
            "show_on_clock": true,
            "forecast_enabled": true,
            "forecast_days": 3,
            "forecast_update": 1
        },
        "wxalert": {
            "alert_feed": "NWS",
            "update_freq": 5,
            "show_alerts": true,
            "nws_show_expire": true,
            "alert_title": true,
            "scroll_alert": true,
            "alert_duration": 5,
            "show_on_clock": true
        }
    },
    "sbio": {
        "screensaver": {
            "enabled": true,
            "animations": true,
            "start": "22:00",
            "stop": "07:00",
            "data_updates": false,
            "motionsensor": false,
            "pin": 7,
            "delay": 30
        },
        "dimmer": {
            "enabled": true,
            "source": "software",
            "daytime": "",
            "nighttime": "",
            "offset": 30,
            "frequency": 5,
            "light_level_lux": 400,
            "mode": "always",
            "sunset_brightness": 5,
            "sunrise_brightness": 40
        },
        "pushbutton": {
            "enabled": false,
            "bonnet": false,
            "pin": 25,
            "reboot_duration": 2,
            "reboot_override_process": "",
            "display_reboot": true,
            "poweroff_duration": 10,
            "poweroff_override_process": "",
            "display_halt": true,
            "state_triggered1": "weather",
            "state_triggered1_process": ""
        }
    }
}

```
**Hardware you are using**
- Raspberry pi model
- The Matrix driver (Adafruit Bonnet or Adafruit HAT)
- Anti-flicker mod or not? 

**Software**
- Which version of the software are you on?

**Comment by falkyre** [comment]
I just downloaded a fresh install of the latest repository (1.6.9) so no config.json was included and I had to run the nhl_setup.  Ran simple setup and the goal_animations has the pref_teams_only: true.

I see in another issue you raised that you were running an older version of the scoreboard.  Did you create the simple config with that version?

**Comment by Overboard-code** [comment]
No. I deleted the whole directory tree. Then did a 'git' then the config
simple.  Then config again. I expected'false' as the value the simple
config had 'true' for me as well.

**Comment by falkyre** [comment]
So when you redid the setup, you selected false for the goal animations?  And it didn't write anything back to the config ?

**Comment by Overboard-code** [comment]
Correct.

**Comment by falkyre** [comment]
Tried 3 times, can't recreate the issue you are seeing.  What I do see from your run of the nhl_setup above is that it looks like the prompt specific to the goal animations did not render correctly.

🥅 Do you want goal animations for only preferred team?

There should be an answer (Yes or No) no matter what (it will default to whatever it pulled from your original config.json)

At this point, I'd chalk it up to one of those un reproducible things.  Are you able to reproduce the issue consistently?

**Comment by Overboard-code** [comment]
Yes I agree.  I ran nhl-setup again and saw:
`🥅 Do you want goal animations for only preferred team? No`
And it worked fin in the config:
```
      "goal_animations": {
            "pref_team_only": false
        }
```
Thanks so much for your time.

**Comment by Overboard-code** [comment]
Found it.  If I just hit Enter instead of 'Y' or 'N' for goal animations it fails  If I enter 'Y' or 'N' on the keyboard it works.  I tried it three times  with the same results.
Hit N and I see:
`🥅 Do you want goal animations for only preferred team? No`
Hit Enter and I see:
`🥅 Do you want goal animations for only preferred team?`

**Comment by falkyre** [comment]
OK,  was able to recreate and fix the problem.  Thanks for finding it.  I'll add the fix and the new binary in a PR for @riffnshred to add to the next release.

**Comment by Overboard-code** [comment]
Thanks so much!  I really appreciate what you are doing.

**Comment by falkyre** [comment]
This is fixed in #377 

@riffnshred will just need to merge it for the next release,

**Comment by falkyre** [comment]
@riffnshred this can be closed now.  It's part of the 1.6.10 version you pushed out last week.
